internal/handlers/tigris/msg_find.go

```diff
@@ -73,18 +73,15 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 		return nil, err
 	}
 
-	firstBatch := types.MakeArray(len(resDocs))
-	for _, doc := range resDocs {
-		firstBatch.Append(doc)
-	}
+	firstBatch, id := common.MakeFindReplyParameters(ctx, resDocs, int(params.BatchSize), fp.DB+"."+fp.Collection)
 
 	var reply wire.OpMsg
 	must.NoError(reply.SetSections(wire.OpMsgSection{
 		Documents: []*types.Document{must.NotFail(types.NewDocument(
 			"cursor", must.NotFail(types.NewDocument(
-				"firstBatch", firstBatch,
-				"id", int64(0), // TODO
+				"id", id,
 				"ns", fp.DB+"."+fp.Collection,
+				"firstBatch", firstBatch,
```

Comment:
From what I can see, MongoDB returns fields in the old order: `firstBatch`, `id`, `ns`. By using the same order we make proxy's diff output much simpler to read

Reply:
Restored order.

Reply:
I remembered why I changed that order. The compatibility test said that `firstBatch` should be in that order.